cli/internal/run/real_run.go

```diff
@@ -89,30 +95,68 @@ func RealRun(
 		Concurrency: rs.Opts.runOpts.Concurrency,
 	}
 
-	mu := sync.Mutex{}
+	logMutex := sync.Mutex{}
+	logWaitGroup := sync.WaitGroup{}
+
+	outputLogsFunc := func(grouped bool, defaultOutWriter io.Writer, defaultErrWriter io.Writer, outBuf bytes.Buffer, errBuf bytes.Buffer) {
```

Comment:
I _think_ we want pointers to the `bytes.Buffer` instances, rather than copies. This doesn't copy the data, but it does copy some metadata included in the `Buffer` struct that I don't _think_ we have a reason to copy.

I also think we can drop the `grouped` bool and only call this in the `grouped` case.